### PR TITLE
chore(dev): update SvelteKit playground to `1.0.0-next.511`

### PR DESCRIPTION
**Pull request by rhinodavid**
## ☕️ Reasoning
[SvelteKit](https://github.com/sveltejs/kit) is rapidly changing as it approaches a 1.0.0 release. This PR updates the SvelteKit PoC app to the latest release and better aligns the app with the demo app created by SvelteKit Create.

## Description
- Update `@sveltejs/kit` from `1.0.0-next.443` to `1.0.0-next.511` (released on 2022-10-06)
- Change app to use [`@sveltejs/adapter-node`](https://kit.svelte.dev/docs/adapters#supported-environments-node-js). This adapter is more appropriate for local development than `@sveltejs/adapter-auto` which was previously used.
- Update [Vite](https://vitejs.dev/) build tool major version (2 -> 3)
- Remove `getFormBody` utility function. This was needed in previous versions of SvelteKit but is no longer required. See the comments of https://dev.to/danawoodman/getting-form-body-data-in-your-sveltekit-endpoints-4a85.
- ~Add `.prettier.rc`. This configuration matches that created by SvelteKit Create with the exception of turning off semicolons to match the existing (questionable) style.~
- Add types where possible/appropriate
- Update README with a small section explaining how to get the app spun up

## Testing

Verified locally. It works!
Screen Shot 2022-10-08 at 2 45 36 PM

## 🧢 Checklist

- [x] Documentation
- [ ] Tests
- [x] Ready to be merged

## 🎫 Affected issues

Please [scout and link issues](https://github.com/nextauthjs/next-auth/issues) that might be solved by this PR.

(Searching open issues with "svelte" or "sveltekit" yields no results)

## 📌 Resources

- [Security guidelines](../Security.md)
- [Contributing guidelines](../CONTRIBUTING.md)
- [Code of conduct](../CODE_OF_CONDUCT.md)
- [Contributing to Open Source](https://kcd.im/pull-request)

### CODE REVIEW

File: apps/playground-sveltekit/package.json [hunk 1]
```diff
@@ -6,14 +6,16 @@
     "dev": "vite dev",
     "build": "vite build",
     "preview": "vite preview",
+    "start": "HOST=127.0.0.1 PORT=5173 ORIGIN=http://localhost:5173 node ./build",
```
Comment (rhinodavid) [inline apps/playground-sveltekit/package.json:9]: matches configuration in `.env.example`

File: apps/playground-sveltekit/.prettierrc [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{
```
Comment (balazsorban44) [inline apps/playground-sveltekit/.prettierrc:1]: We already have Prettier set up for the monorepo, let's keep it consistent. Please remove this file and run prettier on the affected files to apply a consistent linting.

File: apps/playground-sveltekit/README.md [hunk 3]
```diff
@@ -4,6 +4,15 @@ NextAuth.js is committed to bringing easy authentication to other frameworks. ht
...
 SvelteKit support with NextAuth.js is currently experimental. This directory contains a minimal, proof-of-concept application. Parts of this is expected to be abstracted away into a package like `@next-auth/sveltekit`
 
+## Running this Demo
+
+- Copy `.env.example` to `.env`
+- In `.env`, set `GITHUB_CLIENT_ID` and `GITHUB_CLIENT_SECRET`
+  - See [https://docs.github.com/en/developers/apps/building-oauth-apps/creating-an-oauth-app](https://docs.github.com/en/developers/apps/building-oauth-apps/creating-an-oauth-app))
+  - When creating the OAuth app, set "Homepage URL" to `http://localhost:5173` and Authorization callack URL to `http://localhost:5173/api/auth/callback/github`
+- In `.env`, set `NEXTAUTH_SECRET` to any random string
+- Build and run the application: `yarn build && yarn start`
```
Comment (balazsorban44) [inline apps/playground-sveltekit/README.md:14]:
```suggestion
- Build and run the application: `yarn build && yarn start`
```

The monorepo uses pnpm, I think it's time we upgrade this playground, so it relies on the locally built version of NextAuth.js so the dev experience can be enhanced.

**Review by balazsorban44 — CHANGES_REQUESTED** [review]
Thanks, apart from the formatting, the only comment I had is that we should make this playground use pnpm, feel free to address it together with the linting or in a follow-up PR. I am working on some related PRs and will want to come back to this playground in the future, so code-wise I don't have much preference right now, I am happy that it is being experimented on, even in its current state. :pray:

**Comment by rhinodavid** [comment]
@balazsorban44 can you be more specific about the formatting? I ran prettier on my changes (maybe the whole app?). Happy to align this to whatever style you point me to.

File: apps/playground-sveltekit/.prettierrc (hunk 2 above)
Comment (rhinodavid) [reply apps/playground-sveltekit/.prettierrc]: Fixed. The config matched so `format` didn't create any changes.

File: apps/playground-sveltekit/README.md (hunk 3 above)
Comment (rhinodavid) [reply apps/playground-sveltekit/README.md:14]: sounds good. let me get this in a follow-on PR so I can verify nothing breaks with the deps.

**Pull request merged**